src/components/gameScreen/gameScreen.container.js

```diff
@@ -188,38 +166,27 @@ export const GameScreenContainer = () => {
 
   return (
     <GameScreenComponent
-      chosenQuestionsSet={chosenQuestionsSet}
       navigation={navigation}
       questionNumber={questionNumber}
-      currentQuestion={currentQuestion}
-      currentTimeForAnswer={currentTimeForAnswer}
-      timerDuration={timerDuration}
+      currentQuestion={store.currentQuestion.text}
+      currentTimeForAnswer={store.currentQuestion.timeForAnswer}
+      timerDuration={store.currentQuestion.timeForAnswer}
       numberOfQuestions={numberOfQuestions}
       answersAnimation={answersAnimation}
       onTimerAnimationComplete={onTimerAnimationComplete}
       timerColors={timerColors}
       renderAnswerItem={renderAnswerItem}
-      chosenMode={chosenMode}
       timer={timer}
       nextButton={nextButton}
       timerless={timerless}
```

Comment:
I don't see real reason why you should implement render functions for timer and timerless - also these names are not good for this functions (I cannot understand what they mean - timer is usually a variable that stores the ID of the setTimeout/setInterval, timerless - I don't know). Try to get rid of render functions, you are not even wrapping them in useCallbacks so they are just eating your performance